docs/sequence/DFSPOTPValidateModel.puml

```diff
@@ -0,0 +1,19 @@
+@startuml DFSPOTPValidateModel
```

Comment:
I would suggest:
- introducing two final states: success / errored -> where on errored we call  putConsentRequestsError
- introduce transitions for happy flow & corresponding states:  
   - validateOTP (from=start, to=OTPisValid)
   - requestScopes (from=OTPisValid, to=scopesReceived)
   - storeConsent(from=scopesReceived, to=succeed)
   
  Then you can implement this workflow as an extension of PersistenModel class